src/sentry/api/serializers/rest_framework/dashboard.py

```diff
@@ -470,9 +471,34 @@ def validate(self, data):
 
 
 class DashboardPermissionsSerializer(CamelSnakeSerializer[Dashboard]):
-    is_creator_only_editable = serializers.BooleanField(
-        help_text="Whether the dashboard is editable only by the creator.",
+    is_editable_by_everyone = serializers.BooleanField(
+        help_text="Whether the dashboard is editable by everyone.",
     )
+    teams_with_edit_access = serializers.ListField(
+        child=serializers.IntegerField(),
+        help_text="List of team IDs that have edit access to a dashboard.",
+        required=False,
+        default=[],
+    )
+
+    def validate(self, data):
+        if "teams_with_edit_access" in data:
+            team_ids = data["teams_with_edit_access"]
+            existing_team_ids = set(
+                Team.objects.filter(
+                    id__in=team_ids, organization=self.context["organization"]
+                ).values_list("id", flat=True)
+            )
+            invalid_team_ids = set(team_ids) - existing_team_ids
+            if invalid_team_ids:
+                invalid_team_ids_str = ", ".join(str(id) for id in invalid_team_ids)
```

Comment:
There's a util called `oxfordize_list` that we could use here to get a bit of a nicer string representation and avoid having to construct the string explicitly like this